runtime/common/src/lib.rs

```diff
@@ -79,9 +76,31 @@ parameter_types! {
 	/// that combined with `AdjustmentVariable`, we can recover from the minimum.
 	/// See `multiplier_can_grow_from_zero`.
 	pub MinimumMultiplier: Multiplier = Multiplier::saturating_from_rational(1, 1_000_000_000u128);
+	/// Maximum length of block. Up to 5MB.
+	pub BlockLength: limits::BlockLength =
+		limits::BlockLength::max_with_normal_ratio(5 * 1024 * 1024, NORMAL_DISPATCH_RATIO);
+	/// Block weights base values and limits.
+	pub BlockWeights: limits::BlockWeights = limits::BlockWeights::builder()
+		.base_block(BlockExecutionWeight::get())
+		.for_class(DispatchClass::all(), |weights| {
+			weights.base_extrinsic = ExtrinsicBaseWeight::get();
+		})
+		.for_class(DispatchClass::Normal, |weights| {
+			weights.max_total = Some(NORMAL_DISPATCH_RATIO * MAXIMUM_BLOCK_WEIGHT);
+		})
+		.for_class(DispatchClass::Operational, |weights| {
+			weights.max_total = Some(MAXIMUM_BLOCK_WEIGHT);
+			// Operational transactions have an extra reserved space, so that they
+			// are included even if block reached `MAXIMUM_BLOCK_WEIGHT`.
+			weights.reserved = Some(
+				MAXIMUM_BLOCK_WEIGHT - NORMAL_DISPATCH_RATIO * MAXIMUM_BLOCK_WEIGHT,
+			);
+		})
+		.avg_block_initialization(AVERAGE_ON_INITIALIZE_RATIO)
+		.build_or_panic();
 }
 
-const_assert!(AvailableBlockRatio::get().deconstruct() >= AVERAGE_ON_INITIALIZE_WEIGHT.deconstruct());
+const_assert!(NORMAL_DISPATCH_RATIO.deconstruct() >= AVERAGE_ON_INITIALIZE_RATIO.deconstruct());
```

Comment:
can (and should imo) put this in the integrity_test of system. Same potentially applies to all other `const_asserts!` in the root runtime files.

Reply:
What do you mean by `integrity_test`?

Reply:
https://crates.parity.io/frame_support/traits/trait.IntegrityTest.html I recently added examples of this to staking that you can look at. Just adds compile time checks to each pallet that are invoked upon constructing a runtime.

Reply:
Ah, I see. I'm not sure how is this supposed to work in that particular case.
While I was writing a response to this comment I realised that we might be calculating `max_extrinsic` size incorrectly in the original PR and some extra checks might be missing.

It's now updated, and as a consequence this is now validated via `BlockWeights::validate` method. Basically if `AVG_INIT_RATE` is too high, `max_extrinsic` will go down to `0` for some class and it's going to be rejected as invalid configuration.

Reply:
Great, I must admit I was a bit uncertain about ensuring that the `validate` was correct in the original PR. Glad to see this bug is caught here and now.